project/common.scala

```diff
@@ -0,0 +1,159 @@
+import com.typesafe.sbt.SbtPgp.autoImport._
+import org.scalajs.sbtplugin.ScalaJSPlugin.autoImport._
+
+import scala.{ Console => C }
+
+object BuildCommon extends AutoPlugin {
+
+  override def requires = plugins.JvmPlugin && SbtScalariform && HeaderPlugin
+  override def trigger = allRequirements
+
+  object autoImport {
+    lazy val noPublishSettings = Seq(
+      publish := (),
+      publishLocal := (),
+      publishArtifact := false)
+  }
+
+  override def projectSettings =
+    baseSettings ++
+    formatSettings ++
+    enhancingScalaSettings ++
+    publishSettings ++
+    AutomateHeaderPlugin.projectSettings
+
+  private[this] def baseSettings = Seq(
+    scalaVersion := "2.12.0",
+    scalaOrganization := "org.typelevel",
+    crossScalaVersions := Seq("2.11.8", "2.12.0"),
+
+    organization := "com.fortysevendeg",
+    description := "configuration with less hassle",
+
+    fork in run := true,
+    fork in Test := !isScalaJSProject.value,
+    parallelExecution in Test := false,
+    outputStrategy := Some(StdoutOutput),
+    connectInput in run := true,
+    cancelable in Global := true,
+
+    scalacOptions ++= Seq(
+      "-deprecation",
+      "-encoding", "UTF-8",
+      "-feature",
+      "-language:existentials",
+      "-language:higherKinds",
+      "-language:implicitConversions",
+      "-language:experimental.macros",
+      "-unchecked",
+      "-Xfatal-warnings",
+      "-Xlint",
+      "-Yno-adapted-args",
+      "-Ywarn-dead-code",
+      "-Ywarn-numeric-widen",
+      "-Ywarn-value-discard",
+      "-Ywarn-unused-import",
+      "-Xfuture",
+      "-Yno-predef",
+      "-Ypartial-unification"),
+
+    scalacOptions ++= (CrossVersion.partialVersion(scalaVersion.value) match {
+      case Some((2, 11)) => Nil
+      case Some((2, 12)) => Seq("-Yliteral-types")
+      case _             => Nil
+    }),
+
+    libraryDependencies ++= (CrossVersion.partialVersion(scalaVersion.value) match {
+      case Some((2, 11)) => Seq(
+        compilerPlugin("com.milessabin" % "si2712fix-plugin" % "1.2.0" cross CrossVersion.full))
+      case Some((2, 12)) => Nil
+      case _             => Nil
+    }),
+
+    scalacOptions in (Compile, doc) :=
+      (scalacOptions in (Compile, doc)).value.filter(_ != "-Xfatal-warnings"),
+
+    headers := Map(
+      "scala" → (
+        HeaderPattern.cStyleBlockComment,
+        s"""|/* -
+         | * Case Classy [${name.value}]
+         | */
+         |
+         |""".stripMargin))
+  )
+
+  private[this] def enhancingScalaSettings = Seq(
+    //resolvers += Resolver.bintrayRepo("tek", "maven"),
+    //libraryDependencies ++= Seq(
+      //compilerPlugin("tryp" %% "splain" % "0.1.11")),
+    resolvers += Resolver.sonatypeRepo("releases"),
+    libraryDependencies ++= Seq(
+      compilerPlugin(
+        "org.spire-math" %% "kind-projector" % "0.9.3" cross CrossVersion.binary)
+    )
+  )
+
+  private[this] def formatSettings = SbtScalariform.scalariformSettings ++ Seq(
```

Comment:
I was planning on removing Scalariform because it doesn't work well with my pragma tricks (borrowed/adjusted from Doobie).